### PR TITLE
implement s2i testing

### PR DESCRIPTION
**Pull request by TomasTomecek**
You may hit this:
https://bugzilla.redhat.com/show_bug.cgi?id=1510476

This PR also contains:

* add wait() method for containers
* set up logging for tests

### CODE REVIEW

File: conu/backend/docker/image.py [hunk 1]
```diff
@@ -138,3 +139,40 @@ def mount(self, mount_point=None):
...
+        extend this s2i-enabled image using provided source, raises ConuException if
+        `s2i build` fails
+
+        :param source: str, source used to extend the image, can be path or url
+        :param new_image_name: str, name of the new, extended image
+        :param s2i_args: list of str, additional options and arguments provided to `s2i build`
+        :return: S2Image instance
+        """
+        s2i_args = s2i_args or []
+        c = ["s2i", "build"] + s2i_args + [source, self.get_full_name()]
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/docker/image.py]: do you mean to produce a sensible error message if it doesn't?

[the remark this answers is not in the document]
Comment (dhodovsk) [reply conu/backend/docker/image.py]: that, or add s2i to conu requirements.

[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/docker/image.py]: oh right, the requirements.sh script; that's a very good point!

but I'll also make the helper function to check whether the executable is present

**Comment by TomasTomecek** [comment]
I got this error with old s2i:
```
    def test_s2i_usage():
        i = S2IDockerImage(S2I_IMAGE)
>       assert i.usage() == """\
    This is the punchbag S2I image:
    To use it, install S2I: https://github.com/openshift/source-to-image

    Sample invocation:

    s2i build git://<source code> punchbag <application image>

    You can then run the resulting image via:
    docker run <application image>"""
E       assert 'I1108 10:25:...ec/s2i/usage"' == 'This is the p...cation image>'
E         - I1108 10:25:20.673142 00070 install.go:251] Using "usage" installed from "image:///usr/libexec/s2i/usage"
E         + This is the punchbag S2I image:
E         + To use it, install S2I: https://github.com/openshift/source-to-image
E         +
E         + Sample invocation:
E         +
E         + s2i build git://<source code> punchbag <application image>
E         +
E         + You can then run the resulting image via:
E         + docker run <application image>
```

It works well with [this one](https://koji.fedoraproject.org/koji/buildinfo?buildID=995990).

**Pull request merged**